lib/galaxy/webapps/base/controller.py

```diff
@@ -513,8 +513,7 @@ def _copy_hda_to_library_folder(
         ldda.visible = True
         ldda.update_parent_folder_update_times()
         trans.sa_session.commit()
-        ldda_dict = ldda.to_dict()
-        rval = trans.security.encode_dict_ids(ldda_dict)
+        rval = ldda.to_dict()
```

Comment:
That doesn't look right, there's no pydantic model involved here so you should end up with integer database ids